.github/actions/check-downstream-compiles/action.yml

```diff
@@ -57,9 +55,13 @@ runs:
         done <<< "$SUBCRATES"
 
         # Write Git patches for each dependency used downstream
-        for crate in $DEP_NAMES; do
-          crate_path="${subcrates[$crate]}"
-          echo "$crate = { path = \"$crate_path\" }" | tee -a Cargo.toml
+        for file in $(find . -name Cargo.toml -not -path "./target/*"); do
+          !(grep -q "\[patch.'$URL']" $file) && printf "\n\n[patch.'$URL']\n" | tee -a $file
```

Comment:
Why the `!(grep -q "\[patch.'$URL']" $file)` check here? Do we really need it to prevent duplicate `[patch]` sections?

If this `grep` fails and you dont add the `[patch.'$URL']` line, doesn't the `for` loop below (`for crate in $DEP_NAMES`) still run and append the deps to the end of the file regardless? Without that header it might just append to a different unrelated block in `Cargo.toml`, so it might end up making up a nonsensical `Cargo.toml` file I think

Reply:
> Why the !(grep -q "\[patch.'$URL']" $file) check here? Do we really need it to prevent duplicate [patch] sections?

Yes, it will give a duplicate key error.

> If this grep fails and you dont add the [patch.'$URL'] line, doesn't the for loop below (for crate in $DEP_NAMES) still run and append the deps to the end of the file regardless?

Good catch thanks, I've changed it to add the patches directly after the `[patch.'$URL']` line and replace any existing keys to avoid duplicates. It's getting to the point where a proper https://docs.rs/toml/latest/toml/ parser would be easier if more issues crop up.